independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanConfiguratorBase.java

```diff
@@ -278,6 +285,73 @@ public THIS startup() {
         return startup(ObserverMethod.DEFAULT_PRIORITY);
     }
 
+    /**
+     * Declares that an instance of {@code targetClass} may be used as a parameter
+     * to {@link InterceptionProxy#create(Object)} in the {@linkplain BeanCreator creator}
+     * of this synthetic bean.
+     * <p>
+     * The {@code targetClass} is scanned for interceptor binding annotations.
+     *
+     * @param targetClass the target class, must not be {@code null}
+     * @return self
+     */
+    public THIS supportsInterception(Class<?> targetClass) {
```

Comment:
Hm, shouldn't we just use the `BeanConfiguratorBase.implClazz`? I.e. `supportsInterception(DotName... bindingsSourceClasses)`... should be enough.

Reply:
Hmm, yeah, I think this method should be called `injectInterceptionProxy()` or something like that.

It is not super useful I guess, but it allows you to inject 2 interception proxies of 2 different classes and then combine them into a 3rd class, which is the `implClazz` of the synthetic bean. There's a test for this for producers.

Reply:
It's https://github.com/quarkusio/quarkus/pull/41258/files#diff-f0c3315a9b99e04585136501ea4bc529c710bb6f07f1c6729b3ef0888879ccab

Reply:
Renamed to `injectInterceptionProxy()`. Also removed the variants accepting a `DotName` target class; instead, they accept `org.jboss.jandex.Type`.

Reply:
Also removed the `Type` parameter; we now always use the `implClazz` (actually the `providerType`, which is almost always derived from the `implClazz`, but doesn't necessarily have to be).